### ISSUE TITLE
Deferred foreign key constraint doesn't work

### ISSUE THREAD

**Issue by jaroslav-kubicek** [falsecz/easy-pg#9]
Let's suggest that I have table *"segment"* and table *"events"* where each event belongs to segment. Also, "segment" has *"active_event"* field pointing to one event so in that case I have circular dependency, where FK for "active_event" CAN be deferred. 

My problem is that this stopped work when I switched from 0.3.1 to 1.0.2:

```
async.waterfall [
  (cb) -> conn.begin cb
  (res, cb) -> conn.query 'SET CONSTRAINTS segment_active_event_fkey DEFERRED', cb
  (res, cb) -> conn.insert 'segment', {id: 1, name: 'test segment', active_event: 1}, cb
  (res, cb) -> conn.insert 'event', {id: 1, name: 'test event', segment_id: 1}, cb
  (res, cb) -> conn.commit cb
], (err) ->
   ....
````

error: insert or update on table "segment" violates foreign key constraint "segment_active_event_fkey"
detail: 'Key (active_event)=(1) is not present in table "event".'
code: '23503'

is there any reason for this behavior? Thanks!

**Comment by falsecz** [comment]
fixed in v1.0.4

**Comment by jaroslav-kubicek** [comment]
thx :+1:
